openwisp_network_topology/consumers.py

```diff
@@ -2,22 +2,27 @@
 
 from asgiref.sync import async_to_sync
 from channels.generic.websocket import WebsocketConsumer
+from django.core.exceptions import ValidationError
 from swapper import load_model
 
+from .settings import TOPOLOGY_API_AUTH_REQUIRED
```

Comment:
This should be:

```
from settings import app_settings
```

And then use:

```
app_settings.TOPOLOGY_API_AUTH_REQUIRED
```

This allows easier mocking in tests, otherwise the value of  `TOPOLOGY_API_AUTH_REQUIRED` cannot be changed by mocking during a test (it will have no effect).

Please try this, add a test to verify that when ` TOPOLOGY_API_AUTH_REQUIRED` is set to `True`, the websocket consumer allows an unauthenticated user to connect.

Reply:
> Please try this, add a test to verify that when  TOPOLOGY_API_AUTH_REQUIRED is set to True, the websocket consumer allows an unauthenticated user to connect.

If the user is not authenticated and `TOPOLOGY_API_AUTH_REQUIRED` is set to `True`, I think we do not want the user to connect to the WebSocket consumer because the API view displays the message `'Authentication credentials were not provided.'` (as shown below). There is already a test for this present in the code at https://github.com/openwisp/openwisp-network-topology/blob/issue-177/fix-consumer-connection-for-unauthorized-user/openwisp_network_topology/tests/test_websockets.py#L118-L127


![Screenshot from 2023-04-13 23-49-17](https://user-images.githubusercontent.com/56113566/231851604-5d04cf45-dab6-490c-9726-93692b155a29.png)

Reply:
Sorry, I got confused!
I meant to write if `TOPOLOGY_API_AUTH_REQUIRED`, then we shall expect the consumer to be available to anyone.

Reply:
Again, sorry for being always so in a hurry but I have loads of tasks to handle, I meant: if `TOPOLOGY_API_AUTH_REQUIRED` is `False`, assert that the connection to the websocket is public. Is it clear?